### DISCUSSION TITLE
[Feature] Bulk-select images in filtered views the same way as in the unfiltered view.

### DISCUSSION THREAD

**Discussion by x-vibe** [immich-app/immich#9159] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

When bulk-selecting images, you click on one image, hold shift and then click the last image in the selection (Windows File Manager principle and many other OS).

However when you are in a filtered view, you can't select images the same way. Why not, can this be added? It's really manual to add pictures to albums, once you are in the "filter out albums" view... 

Cheers, your program is awesome, I love it!

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by klendgit** [comment]
Applies also to the Map interface: you can drill down to a location and click on the images, but not selecting them. I would be in favor to show a thumbnail viewer in the case when the 'bullet' contains more than one image (or a config setting like: 'show thumbnail viewer in filtered view when selection contains more than 1 picture').

**Comment by scoop96** [comment]
This would be awesome, way easier to clean the big uploads from the PC to the server <3

**Comment by maxshcherbina** [comment]
Just checked in on this - still would be a welcome addition to the app :)

**Comment by petersuba** [comment]
Since #12263 was closed, adding it here as a reference to make sure folder view is considered (I don't know if technically that is also a "filtered view" or not, just making sure)

**Reply by nielstron** [reply]
Looking at the folder view, I notice that "select assets" is turned off: https://github.com/immich-app/immich/blob/944ea7dbcdafe6652bb4d9f632a45c7d1916b184/web/src/routes/(user)/folders/%5B%5Bphotos%3Dphotos%5D%5D/%5B%5BassetId%3Did%5D%5D/%2Bpage.svelte#L84
Is there any particular reason for this? What needs to be done besides turning this on to allow selecting multiple assets?

Looking at the search view that seems to have similar structure, at least a header bar is needed to show edit options
https://github.com/immich-app/immich/blob/944ea7dbcdafe6652bb4d9f632a45c7d1916b184/web/src/routes/(user)/search/%5B%5Bphotos%3Dphotos%5D%5D/%5B%5BassetId%3Did%5D%5D/%2Bpage.svelte#L231

**Comment by DaanHend** [comment]
Notation of closed issue #13001. 

Would help the consistency through the app of certain actions.
Nice feature continuation of the feature implemented in #3086

**Comment by scoop96** [comment]
@DaanHend  is not fixed... may you check the answer from @nielstron  I was following this issue from some time.

here is a proof that does not work if I Shift+select doesnt work as intended

![image](https://github.com/user-attachments/assets/0c99f476-bc41-428c-8166-086760d46761)

**Comment by rholloway** [comment]
Would love this feature. I'm finding that the photos timeline view I can do a lot more and most of the features I'm looking for are there from grouping/selection to add to albums and tags (select all button, shift click to group a range, etc.). However, as soon as I do a search to restrict photos (tens of thousands of photos it's almost necessary) or view from folders (where they are external library and were pre-organized in folder structure prior to immich), a lot of the features disappear that make it easy to re-organize things in immich (album/tagging in bulk)

**Comment by TildenWinston** [comment]
Worth noting that while shift select does now work in the filtered or search view, it doesn't work on the default photos view

**Comment by DaanHend** [comment]
@TildenWinston & @rholloway 

This 'feature' is added in #14213. And now available with V1.121.0

https://github.com/immich-app/immich/releases/tag/v1.121.0

**Comment by nielstron** [comment]
I suspect the closure of this issue was a mistake by @bo0tzz . I just updated my server to check and indeed the folder view does still not support bulk-selection of images. @DaanHend s comment was referring to a small fix that only added shift+select to views where bulk select was already supported.

If it was not a mistake please allow at least one open feature request for bulk-select in folder view without immediately closing it. The issue was raised several times, every time closed, sometimes referred here.

**Reply by DaanHend** [reply]
This ticket is related to multi selection in filter views. Where this was not yet supported before 1.121.0. 
The folder view would be another feature request.

That is how I interpreted the whole ticket as it also says in the title

**Reply by nielstron** [reply]
I am aware, however there is a history of feature requests for multi-select in folder views (i.e. https://github.com/immich-app/immich/discussions/12263 by @petersuba , https://github.com/immich-app/immich/discussions/14132, https://github.com/immich-app/immich/discussions/14118) which are closed with reference to this discussion. @bo0tzz please fix this.

**Reply by petersuba** [reply]
Agree @nielstron Not all of us are aware of the nuances of which views/components of the solution are relying on which elements of the code. I for one am trying to follow the closures and cross-reference such that the completeness of the discussion is not lost when something is closed due to claimed duplication. I think either close an issue only when its full scope is complete, or slice off parts of the discussion into separate threads before closing it. Comments are frequently made on threads when non-related aspects enter into the conversation that those should be separate threads, and that's fine. But if one wishes to just close an item to reflect progress and there are substantial scope remains, I think it would be appreciated if those are then added for us in separate thread. It is quite frustrating having to chase after our thoughts, find out that closure didn't really address those, then being told that those should be separate items, then create them, with the risk that we'll be in a similar situation down the line. I appreciate that development sometimes branches out, addressing parts of discussions etc and more than willing to help, but please also reciprocate this and consider the full thread and process closure actions accordingly. 

Just to add that if a feature is interpreted solely based on what is in the title, then I am not sure why are we discussing items, when it could just be a back-and forth between the requester and the developers, however in that case be prepared to have a LOT more feature requests/bug reports with many partially overlapping scope elements as these normally develop through the discussion.

**Reply by nielstron** [reply]
@bo0tzz to summarize the above comment
- the title of this feature request now does not match the remaining open issue (multi-select in folder view) anymore
- the main concern has been resolved (shift+click multi-select in other views), closing this discussion is good to show progress
- a new feature request for multi-select in folders should be opened to clarify the remaining scope

Please also provide some reasoning when closing/opening threads, as without reasoning the experience for well-meaning feedback-giving users like us is very frustrating.

**Reply by bo0tzz** [reply]
:+1: please open a new thread for the folder view.

**Comment by nielstron** [comment]
for everyone following this thread for Multi-Select on folder views, I created a seperate discussion for it. Please upvote https://github.com/immich-app/immich/discussions/14307
